### PR TITLE
Add information on using an automation to renew a certificate

### LINKED ISSUE #2423: Let's Encrypt add-on terminates process and will not auto renew unless add-on manually re-started
### Describe the issue you are experiencing

I have version 4.12.0 on the Let's Encrypt add-on installed. This is set to "start on boot". Looking at the log files I can see that the add-on starts, checks the current cert files (renews if necessary), then the process is terminated. The result is that I need to manually start the add-on again when I'm within 30 days of my certificate expiring. 

I would expect the add-on to remain running and periodically check the certs and renew if necessary. Is there something I have configured incorrectly? Or is this by design, or a bug?

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

Let's Encrypt

### What is the version of the add-on?

4.12.0

### Steps to reproduce the issue

1. Start the add-on
2. View the logs 
3. See that the final log message contains "sending all processes the KILL signal and exiting."
[Log.txt](https://github.com/home-assistant/addons/files/8372980/Log.txt)

### Anything in the Supervisor logs that might be useful for us?

```txt
22-03-29 08:19:05 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-03-29 08:42:34 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING
22-03-29 08:42:34 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.PLUGIN
22-03-29 08:42:34 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE
22-03-29 08:42:34 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM
22-03-29 08:42:34 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.CORE
22-03-29 08:42:34 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON
22-03-29 08:42:34 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.SUPERVISOR
22-03-29 08:42:34 INFO (MainThread) [supervisor.resolution.check] System checks complete
22-03-29 08:42:34 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING
22-03-29 08:42:35 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
22-03-29 08:42:35 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state CoreState.RUNNING
22-03-29 08:42:35 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
22-03-29 08:44:45 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/home-assistant/addons repository
22-03-29 08:44:45 INFO (MainThread) [supervisor.store] Loading add-ons from store: 23 all - 0 new - 0 remove
22-03-29 08:44:45 INFO (MainThread) [supervisor.store] Loading add-ons from store: 23 all - 0 new - 0 remove
22-03-29 08:49:05 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-03-29 09:19:05 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.PLUGIN
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.CORE
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.SUPERVISOR
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.check] System checks complete
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state CoreState.RUNNING
22-03-29 09:42:35 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
22-03-29 09:43:55 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json
22-03-29 09:49:06 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-03-29 09:59:07 INFO (MainThread) [supervisor.host.info] Updating local host information
22-03-29 09:59:07 INFO (MainThread) [supervisor.host.services] Updating service information
22-03-29 09:59:07 INFO (MainThread) [supervisor.host.network] Updating local network information
22-03-29 09:59:08 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information
22-03-29 09:59:08 INFO (MainThread) [supervisor.host.manager] Host information reload completed
22-03-29 10:19:06 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.PLUGIN
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.CORE
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.SUPERVISOR
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.check] System checks complete
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state CoreState.RUNNING
22-03-29 10:42:35 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
22-03-29 10:49:07 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-03-29 11:19:07 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-03-29 11:19:57 INFO (SyncWorker_1) [supervisor.docker.interface] Cleaning addon_core_letsencrypt application
22-03-29 11:19:58 INFO (SyncWorker_1) [supervisor.docker.addon] Starting Docker add-on homeassistant/aarch64-addon-letsencrypt with version 4.12.0
22-03-29 11:21:56 INFO (SyncWorker_3) [supervisor.docker.interface] Cleaning addon_core_letsencrypt application
22-03-29 11:21:56 INFO (SyncWorker_3) [supervisor.docker.addon] Starting Docker add-on homeassistant/aarch64-addon-letsencrypt with version 4.12.0
22-03-29 11:22:20 ERROR (SyncWorker_7) [supervisor.docker.interface] Container addon_core_letsencrypt is not running
```

### Anything in the add-on logs that might be useful for us?

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Twincarb**
To save having to trawl through forums to find out how to automate the renewal of a certificate it makes sense to me to add the information to the documentation.

Fixes #2423

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @Twincarb, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;addons#2489).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Review by mdegat01 — CHANGES_REQUESTED** [review]
I don't mind this but I think it would be better to make a blueprint with an input for `time`. That way we can just include an "add blueprint" button in the doc and people can add it.

Do you want to make that blueprint? Or i can if you prefer. I doubt there will be much maintenance (if any) maintenance required for it.

File: letsencrypt/DOCS.md [hunk 1]
```diff
@@ -461,6 +461,22 @@ dns-gandi
 dns-transip
 ```
 
+## Auto update with an automation
```
Comment (frenck) [inline letsencrypt/DOCS.md:464]: We should make this a Blueprint instead...
That makes is importable with a single button.

**Comment by Twincarb** [comment]
It makes sense to me to use a Blueprint, I have just created my first Blueprint on my system to test it works. Where should it be published too?  https://community.home-assistant.io/c/blueprints-exchange 
To address @ludeeus point that HA requires restarting, is this something that can be added as an action to the Blueprint with a rule so it only restarts on the certificate being successfully being renewed?

**Comment by mdegat01** [comment]
@Twincarb Yes it should be published to that section of the forum you linked. See [sharing blueprints](https://community.home-assistant.io/t/about-blueprints/253788#sharing-blueprints-2) in the about blueprints post in that section. Basically make a post, put your blueprint in, generate a badge and then add that badge to the doc here instead of the YAML. That way its just click button to import and fill in the time to run at.

**Comment by elahd** [comment]
@mdegat01 IMO most people using this addon want to set-and-forget. That is, they install the add-in, configure it once, and then just trust it to renew for them. Using a blueprint increases complexity.

As a compromise here, why not make this a configuration option for the addon? Changes would be, roughly:

1. Add a "Renew Automatically" configuration option to the addon that is on by default for new installs.
2. Change addon's `startup` attribute from `manual` to `application` so that it auto runs when HA starts up.
3. In `addons/letsencrypt/rootfs/etc/services.d/lets-encrypt`, rename `run` to `renew`.
4. Create new run script that runs the following workflow:
a. Execute `renew` script to renew certificate when addon starts.
b. If "Renew Automatically" is **on**, create a cron job that runs `renew` every Sunday at midnight. The addon should stay running.
c. If "Renew Automatically" is **off**, the addon exits as it does today.

This way, we can accommodate both what I think is the most common workflow by default while still letting edge case users do whatever they do today.

Also, since this addon is often paired with the NGINX Home Assistant SSL Proxy, we should do **_one_** of the following:
1. Add an off-by-default configuration option to the Let's Encrypt addon to automatically restart NGINX Home Assistant SSL Proxy after certificate renewal.
2. Modify NGINX Home Assistant SSL Proxy to restart after it detects that the SSL certs have been renewed. (Easiest way is to look for updates to the cert file itself; not by monitoring other services.)

Thoughts on this? I'm happy to do the work, but only want to do it if the PR will be approved 😀.

**Comment by elahd** [comment]
What's the larger philosophy around what should be a blueprint vs what should be core functionality? IMO if _everyone_  who uses the add-on needs to install this blueprint in order for the addon to work as expected, the behavior should just be rolled into the addon.

It would be easier for everyone if the integration just checked the SSL cert's expiration date nightly and renewed as needed.

I proposed a more detailed proposal at https://github.com/home-assistant/addons/issues/2423#issuecomment-1187695582 but haven't gotten feedback from maintainers.

**Pull request closed without merging**